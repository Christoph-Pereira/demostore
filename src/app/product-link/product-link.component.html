<a routerLink="{{url}}">
    <div class="productLinkDiv" [ngStyle]="{backgroundImage: 'url(' + image + ')'}">
        <p class="productLinkText">
            {{text}}
        </p>
    </div>
</a>