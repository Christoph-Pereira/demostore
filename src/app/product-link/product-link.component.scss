a {
    text-decoration: none;
}
.productLinkDiv {
    height: 450px;
    width: 685px;
    margin: auto;
    transition: all .2s ease-in-out;

    .productLinkText {
        font-size: 2rem;
        padding: 2rem;
        color: white;
    }
}

.productLinkDiv:hover {
    transform: scale(1.1);
}