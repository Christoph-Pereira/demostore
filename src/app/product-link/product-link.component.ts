import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'app-product-link',
  templateUrl: './product-link.component.html',
  styleUrls: ['./product-link.component.scss']
})
export class ProductLinkComponent {
  
  @Input() public text: String | undefined;
  @Input() public image: String | undefined;
  @Input() public url: String | undefined;

  constructor() { }

}
