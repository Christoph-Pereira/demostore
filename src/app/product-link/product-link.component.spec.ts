import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ProductLinkComponent } from './product-link.component';
import { RouterTestingModule } from '@angular/router/testing';

describe('ProductLinkComponent', () => {
  let component: ProductLinkComponent;
  let fixture: ComponentFixture<ProductLinkComponent>;
  const productLink = {
        "text": "Advanced Gamification",
        "image": "../../assets/images/Centrical-background-2.png",
        "url": "/product/1"
      };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ProductLinkComponent ],
      imports:[ RouterTestingModule ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ProductLinkComponent);
    component = fixture.componentInstance;
    
    component.text = productLink.text;
    component.image = productLink.image;
    component.url = productLink.url;

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render with image', () => {
    expect(fixture).toMatchSnapshot();
  });
});