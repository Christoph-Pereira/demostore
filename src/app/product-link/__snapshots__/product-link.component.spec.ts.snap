// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductLinkComponent should render with image 1`] = `
<app-product-link
  __ngContext__={[Function LRootView]}
  image={[Function String]}
  text={[Function String]}
  url={[Function String]}
>
  <a
    href="/product/1"
  >
    <div
      class="productLinkDiv"
      style="background-image: url(../../assets/images/Centrical-background-2.png);"
    >
      <p
        class="productLinkText"
      >
         Advanced Gamification 
      </p>
    </div>
  </a>
</app-product-link>
`;
